App crashed while enabling EOS token

Crash log:

 ```
       Caused by: java.lang.ArrayIndexOutOfBoundsException: length=11; index=-1
        at java.util.Arrays$ArrayList.get(Arrays.java:3769)
        at io.horizontalsystems.bankwallet.modules.balance.BalanceModule$BalanceItemDataSource.setBalance(BalanceModule.kt:118)
        at io.horizontalsystems.bankwallet.modules.balance.BalancePresenter.didUpdateBalance(BalancePresenter.kt:128)
        at io.horizontalsystems.bankwallet.modules.balance.BalanceInteractor.subscribeToAdapterUpdates(BalanceInteractor.kt:90)
        at io.horizontalsystems.bankwallet.modules.balance.BalanceInteractor.access$subscribeToAdapterUpdates(BalanceInteractor.kt:10)
        at io.horizontalsystems.bankwallet.modules.balance.BalanceInteractor$initWallets$2.accept(BalanceInteractor.kt:40)
        at io.horizontalsystems.bankwallet.modules.balance.BalanceInteractor$initWallets$2.accept(BalanceInteractor.kt:10)
```

--- a/app/src/main/java/io/horizontalsystems/bankwallet/modules/balance/BalanceInteractor.kt
+++ b/app/src/main/java/io/horizontalsystems/bankwallet/modules/balance/BalanceInteractor.kt
@@ -71,6 +71,7 @@ class BalanceInteractor(
         delegate?.didUpdateCurrency(currencyManager.baseCurrency)
     }
 
+    @Synchronized
     private fun onUpdateWallets() {
         adapterDisposables.clear()
 
@@ -83,6 +84,7 @@ class BalanceInteractor(
         }
     }
 
+    @Synchronized
     private fun subscribeToAdapterUpdates(wallet: Wallet, initialUpdate: Boolean) {
         adapterManager.getBalanceAdapterForWallet(wallet)?.let { adapter ->
 

